backend/src/main/resources/schema.graphqls

```diff
@@ -60,25 +66,26 @@ type Patient {
   middleName: String
   lastName: String
   suffix: String
-  race: String
+  race: String @requiredPermissions(allOf: ["READ_PATIENT_LIST"])
   birthDate: LocalDate
```

Comment:
I ended up removing the `type`-level directives because they don't play well with enforcing gates through data fetchers.

A query like [this test case](https://github.com/CDCgov/prime-simplereport/blob/bf5b2a264ade53be37a53377391b39854bbfaa8f/backend/src/test/resources/person-with-last-test-result-query) will hit three permissions gates:

1. One at the query level, where the data fetcher will see that users need `anyOf: [READ_PATIENT_LIST, SEARCH_PATIENTS]` to access the `patients` node of the `Query` type.
1. One at `Patient::telephone`, where the data fetcher will see that users need `anyOf: [READ_PATIENT_LIST, UPDATE_TEST]` to access the field.
1. One at `Patient::lastTest`, where the data fetcher will see that users need `anyOf: [READ_PATIENT_LIST, UPDATE_TEST]` to access the field.

GraphQL schema wiring only lets you set up a proxy data fetcher on fields that have the directive, so there's no opportunity to make a gate like "users must have `anyOf: [READ_PATIENT_LIST, SEARCH_PATIENTS]` to access any property of the `Patient` type" (even though that's an assertion we'd like to make!) so I'm just relying on a directive on the `patients` query to simulate that.

Reply:
So it turns out this is possible, though a bit verbose. I'll add some commentary on the schema to explain how it works.

Reply:
@TomNUSDS I added object-level directives and some documentation; would you mind taking another look?